src/modules/taskStore.js

```diff
@@ -0,0 +1,53 @@
+import Storage from './storage.js';
+import Task from './task.js';
+
+const ITEM_STORAGE = 'toDoTasks';
+/**
+ * Class to Add, Remove and Get tasks from Storage from
+ */
+export default class TaskStore {
+  tasksCollection = [];
+
+  taskStorage;
+
+  constructor() { this.taskStorage = new Storage(ITEM_STORAGE); }
+
+  add= (description) => {
+    this.tasksCollection = this.taskStorage.getItemStorage();
+    const id = (this.tasksCollection !== null) ? this.tasksCollection.length + 1 : 1;
+    const objectTask = new Task(id, false, description);
+    this.taskStorage.addItemStorage(objectTask);
+  }
+
+  remove=(id) => {
+    if (id > 0) {
+      this.tasksCollection = this.taskStorage.getItemStorage();
+      const resultCollection = this.tasksCollection.filter((t) => t.index !== parseInt(id, 10));
+      let newId = 1;
+      // reindexation
+      resultCollection.forEach((element) => {
+        element.index = newId;
+        newId += 1;
+      });
+      this.taskStorage.removeItemStorage(resultCollection);
+    }
+  }
```

Comment:
- [ ] Kindly your `Clear all completed` method works fine, but you also implement the delete for every specific task, I have tried to delete but it just reload the page nothing got removed from the page or the local storage.

Talking about this button
![image](https://user-images.githubusercontent.com/35535130/160967703-098a191f-4d9f-4d37-9106-14e4fc4008ec.png)

Reply:
Hi @keroloussamy: As it appears in the video "To-Do list", there are 2 operations associated with editing: 

1. Changing the text and leaving the textbox triggers the update event.
2. If all text is deleted from the textbox and the trash icon is clicked, the task is removed from the list. (I believe that the removal is being carried out correctly following this process. )

Reply:
Hello @NeckerFree, The delete method should be working when you click on the trash icon, in the video it just show you that you can edit and remove, but it doesn't mean that you have to remove text first in order to remove.

Reply:
Ok @keroloussamy: I have included the option to delete when the textbox does not change. 
If the text has changed and the user clicks on the trash can icon, the update event is activated and the deletion event is canceled.

Reply:
@NeckerFree  So do you mean that if the text has changed and the user clicks on the trash icon, it won't be deleted ?
if that's what do you mean. So no, the task should be deleted whenever the trash icon is clicked.

Reply:
Yes, @keroloussamy because I need to fire the update event when the text has changed. 
And if you see the reference to the prototype app ([web.archive.org]) in [the big picture of this project](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/sneak_peek.md) my application is working in the same way.